### PR TITLE
Add Frame-VAD examples and utils

### PR DESCRIPTION
**Pull request by stevehuang52**
# What does this PR do ?

The second PR of Frame-VAD. Please merge the first PR first:  https://github.com/NVIDIA/NeMo/pull/6441

**Collection**: [ASR]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

### CODE REVIEW

File: nemo/collections/asr/data/audio_to_label.py
Comment (tango4j) [inline nemo/collections/asr/data/audio_to_label.py:897]: spliting ->splitting

File: nemo/collections/asr/data/audio_to_label.py
Comment (tango4j) [inline nemo/collections/asr/data/audio_to_label.py:1108]: spliting ->splitting

File: examples/asr/speech_classification/frame_vad_infer.py
Comment (tango4j) [inline examples/asr/speech_classification/frame_vad_infer.py:207]: I strongly suggest the line from 207 to 245 be a separate function and put it into a new python file called `NeMo/nemo/collections/asr/metrics/det_er.py`

For example, all diarization related metric functions and evaluation functions are in 
https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/asr/metrics/der.py

**Review by tango4j — CHANGES_REQUESTED** [review]
Two major points:
(1) missing unit tests for util file.
there are many newly created functions in vad_utils.py which are missing unit-tests.
It's better to make unit-tests for them to maintain the code base easily by spotting the errors.
(2) Functionization of example script
It would be desirable to convert the lines in example file to functions in vad_utils.py.
Also unit-tests for these functions should be created.

File: nemo/collections/asr/parts/preprocessing/segment.py
Comment (tango4j) [inline nemo/collections/asr/parts/preprocessing/segment.py:190]: The normalization feature must have thorough unit tests since `segment.py` is used by all ASR related NeMo modules. 

Be sure to add normalization tests and also, it needs to handle multi-channel capability to normalize multiple files together.

File: nemo/collections/asr/parts/utils/vad_utils.py
Comment (tango4j) [inline nemo/collections/asr/parts/utils/vad_utils.py:1000]: [Comment ID: a03 - Please change these lines to use `AudioSegment.from_file()`]
Please replace 1000~1004 with `AudioSegment.from_file()`.

File: nemo/collections/asr/parts/utils/vad_utils.py
Comment (tango4j) [inline nemo/collections/asr/parts/utils/vad_utils.py:1473]: Please see [Comment ID: a03 - Please change these lines to use `AudioSegment.from_file()`]

**Review by tango4j — CHANGES_REQUESTED** [review]
+ Please make sure all audio file reading in `vad_utils` use `AudioSegment.from_file()` since it controls all kinds of multi-channel file loading and offset/duration handling. This is required for the upcoming multi-channel based ASR and diarization.

File: nemo/collections/asr/parts/preprocessing/segment.py
Comment (tango4j) [inline nemo/collections/asr/parts/preprocessing/segment.py:467]: Please see [Comment ID: a05 - multichannel feature for normalization]

File: nemo/collections/asr/parts/utils/vad_utils.py
Comment (stevehuang52) [reply nemo/collections/asr/parts/utils/vad_utils.py:1000]: we don't need AudioSegment for visualization, librosa is fine

File: examples/asr/conf/vad/frame_vad_infer_postprocess.yaml
Comment (fayejf) [inline examples/asr/conf/vad/frame_vad_infer_postprocess.yaml:22]: These are not tuned parameters, right? If we don't have a plan to tune them, how about leaving a note here?

File: examples/asr/speech_classification/frame_vad_infer.py
Comment (fayejf) [reply examples/asr/speech_classification/frame_vad_infer.py:175]: agree. let's try moving the functions to util files to simply the script and for reusability. such as `eval_detection_error`

File: examples/asr/speech_classification/frame_vad_infer.py
Comment (fayejf) [inline examples/asr/speech_classification/frame_vad_infer.py:192]: and function like this could be moving to vad_utils

File: examples/asr/speech_classification/speech_to_frame_label.py
Comment (fayejf) [inline examples/asr/speech_classification/speech_to_frame_label.py:50]: from a pre-trained checkpoint

File: examples/asr/speech_classification/frame_vad_infer.py
Comment (stevehuang52) [reply examples/asr/speech_classification/frame_vad_infer.py]: done

File: examples/asr/conf/vad/frame_vad_infer_postprocess.yaml
Comment (stevehuang52) [reply examples/asr/conf/vad/frame_vad_infer_postprocess.yaml:22]: done

**Review by fayejf — APPROVED** [review]
LGTM. Let's wait for @tango4j's approval too.

File: nemo/collections/asr/data/audio_to_label.py
Comment (stevehuang52) [reply nemo/collections/asr/data/audio_to_label.py]: fixed

File: nemo/collections/asr/data/audio_to_label.py
Comment (stevehuang52) [reply nemo/collections/asr/data/audio_to_label.py:897]: fixed

File: nemo/collections/asr/data/audio_to_label.py
Comment (stevehuang52) [reply nemo/collections/asr/data/audio_to_label.py]: fixed

File: nemo/collections/asr/data/audio_to_label.py
Comment (stevehuang52) [reply nemo/collections/asr/data/audio_to_label.py]: fixed

File: nemo/collections/asr/models/classification_models.py
Comment (stevehuang52) [reply nemo/collections/asr/models/classification_models.py]: removed

File: nemo/collections/asr/models/classification_models.py
Comment (stevehuang52) [reply nemo/collections/asr/models/classification_models.py]: done

File: examples/asr/speech_classification/frame_vad_infer.py
Comment (stevehuang52) [reply examples/asr/speech_classification/frame_vad_infer.py]: done

File: nemo/collections/asr/models/classification_models.py
Comment (stevehuang52) [reply nemo/collections/asr/models/classification_models.py]: done

File: tests/collections/asr/test_asr_classification_model.py
Comment (stevehuang52) [reply tests/collections/asr/test_asr_classification_model.py]: done

File: nemo/collections/asr/parts/utils/vad_utils.py
Comment (stevehuang52) [reply nemo/collections/asr/parts/utils/vad_utils.py]: done

File: nemo/collections/asr/parts/preprocessing/segment.py
Comment (stevehuang52) [reply nemo/collections/asr/parts/preprocessing/segment.py]: done

File: nemo/collections/asr/parts/preprocessing/segment.py
Comment (stevehuang52) [reply nemo/collections/asr/parts/preprocessing/segment.py:291]: done

File: nemo/collections/asr/parts/utils/vad_utils.py
Comment (stevehuang52) [reply nemo/collections/asr/parts/utils/vad_utils.py]: done

File: nemo/collections/asr/parts/utils/vad_utils.py
Comment (stevehuang52) [reply nemo/collections/asr/parts/utils/vad_utils.py:1067]: not sure how to test plots, let's just leave it

File: nemo/collections/asr/parts/utils/vad_utils.py
Comment (stevehuang52) [reply nemo/collections/asr/parts/utils/vad_utils.py]: done

**Comment by tango4j** [comment]
Seems like most of the comments are addressed, and the unimplemented ones are not so crucial ones. I think we can merge after the CI passes.

**Review by tango4j — APPROVED** [review]
Very significant contribution on VAD pipeline. Thanks a lot. Approving.

**Pull request merged**
